apisix/utils/log-util.lua

```diff
@@ -18,6 +18,22 @@ local core     = require("apisix.core")
 
 local _M = {}
 
+
+local function remove_stale_objects(premature, log_buffer, status)
+    if premature then
+        return
+    end
+
+    for key, batch in ipairs(log_buffer) do
+        if #batch.entry_buffer.entries == 0 and #batch.batch_to_process == 0 then
+            core.log.debug("removing batch processor stale object, route id:" .. tostring(key))
+            log_buffer[key] = nil
```

Comment:
Can we delete stale objects here?

https://github.com/apache/incubator-apisix/blob/master/apisix/plugins/kafka-logger.lua#L120

Reply:
yes, this should be moved inside each plugin itself.